src/env/env.c

```diff
@@ -0,0 +1,94 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   env.c                                              :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: alopez-g <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/10/15 18:07:51 by alopez-g          #+#    #+#             */
+/*   Updated: 2022/10/16 01:18:23 by alopez-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "env.h"
+
+//easy version
+void	env_unsetvar(char ***env, char *key)
+{
+	int		i;
+	int		esize;
+
+	i = 0;
+	esize = env_size(*env);
+	while (*(*env + i))
+	{
+		if (key_compare(*(*env + i), key))
+		{
+			free(*(*env + i));
+			*(*env + i) = *(*env + esize - 1);
+			*(*env + esize - 1) = NULL;
+			return ;
+		}
+		i++;
+	}
+}
+
+char	*env_getvar(char ***env, char *key)
+{
+	while (**env)
+	{
+		if (key_compare(**env, key))
+			return (get_value(**env));
+		(*env)++;
+	}
+	return (NULL);
+}
+
+t_error_code	env_setvar(char ***env, char *key, char *value)
+{
+	char	i;
+
+	i = 0;
+	if (*env)
+	{
+		while (*(*env + i))
+		{
+			if (key_compare(*(*env + i), key))
+			{
+				free(*(*env + i));
+				*(*env + i) = get_key_value_pair(key, value);
+				return (SUCCESS);
+			}
+			i++;
+		}
+	}
+	new_var(env, key, value);
+	return (SUCCESS);
+}
+
+t_error_code	env_initenv(char ***env, char **envp)
```

Comment:
shoule we do something with SH_LVL?
i think it goes up evey time we initenv

Reply:
I don't know what you mean by this

Reply:
Like a depth LVL of how many cmds we created?